src/components/v2v3/V2V3Project/V2V3PayButton/hooks/useV2V3PayButton.test.tsx

```diff
@@ -0,0 +1,202 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { act, renderHook } from '@testing-library/react-hooks'
+import {
+  PayProjectFormContext,
+  PayProjectFormContextType,
+} from 'components/Project/PayProjectForm/payProjectFormContext'
+import {
+  ProjectMetadataContext,
+  ProjectMetadataContextType,
+} from 'contexts/shared/ProjectMetadataContext'
+import {
+  V2V3ProjectContext,
+  V2V3ProjectContextType,
+} from 'contexts/v2v3/Project/V2V3ProjectContext'
+import { V2V3_CURRENCY_ETH } from 'utils/v2v3/currency'
+import { useV2V3PayButton } from './useV2V3PayButton'
+
+type DeepPartial<T> = {
+  [P in keyof T]?: T[P] extends Array<infer U>
+    ? Array<DeepPartial<U>>
+    : T[P] extends ReadonlyArray<infer U>
+    ? ReadonlyArray<DeepPartial<U>>
+    : T[P] extends object | undefined
+    ? DeepPartial<T[P]>
+    : T[P]
+}
+
+const withContextPropsWrapper = ({
```

Comment:
We can move this and make it a bit more generic, I just left it here for the POC pr